### PR TITLE
DENG-6159 Add Ray Serve example to demonstrate batching feature

### PR DESCRIPTION
**Pull request by abhi-agg**
### Goal(s) of this Pull Request:

- The Translation team has expressed interest for one of their projects (related to WebCompat) to use batching for (LLM based) inference server in order to optimized the hardware usage and reduce compute costs for the inference.

As a prepration for onboarding them, here is an example demonstrating Ray Serve's dynamic request batching feature.

### Example of how it works:

... Added README to explain how it works.

### Acceptance criteria:

In order to approve, reviewer must be able to...

- [x] ...

### CODE REVIEW

File: examples/ray_serve/batched_translator/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    > [!NOTE]
+    > You can modify the [serve_config.yaml](./serve_config.yaml) file to set the [batching parameters](https://docs.ray.io/en/latest/serve/api/doc/ray.serve.batch.html#ray-serve-batch) `max_batch_size` and `batch_wait_timeout_s` as per your requirement
+
+    ```sh
+    serve run serve_config.yaml
+    ```
+4. From separate terminals, let multiple clients call the locally running service endpoint and check if it returns the expected response
+   > [!NOTE]
+   > You can use the following curl command to send a request (you can modify the text to be translated for each request)
```
Comment (chelseatroy) [inline examples/ray_serve/batched_translator/README.md:27]: I think it might also be worth mentioning that you can go to http://127.0.0.1:8000/docs and run a request in the interactive request console in the browser itself.

File: examples/ray_serve/batched_translator/batched_translator.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            user_config["batch_wait_timeout_s"]
+        )
+
+
+# Ray Serve Application builder
+def batched_translator_app_builder(args: BatchedTranslatorArgs) -> Application:
+    return BatchedTranslator.bind(args.task, args.model)
+
+
+# Un-comment the following line ONLY to auto-generate Serve config file using `serve build` command and then comment it back again
```
Comment (chelseatroy) [inline examples/ray_serve/batched_translator/batched_translator.py:83]: Maybe worth explicitly mentioning when one might want/need to run that command? Or at least that consumers of this example _don't_ need to run it.

**Review by chelseatroy — COMMENTED** [review]
Few small comments added! Great work!

File: examples/ray_serve/batched_translator/batched_translator.py (hunk 2 above)
Comment (abhi-agg) [reply examples/ray_serve/batched_translator/batched_translator.py:83]: @chelseatroy I realized that I didn't frame it properly. Let me rephrase it to make it explicit.

**Review by chelseatroy — APPROVED** [review]
